receiver/prometheusoperatorreceiver/DESIGN.md

```diff
@@ -0,0 +1,142 @@
+## Design Goals
+
+### Vision
+Provide a nearly zero-config migration path from PrometheusOperator defined scraping configs.  
+
+### Scope
+Only scrape config relevant CRDs (CustomResourceDefinition) are in scope of this receiver.    
+As of this writing the relevant CRDs are: 
+- ServiceMonitor
+- PodMonitor
+
+The receiver should be able to query this CRDs from Kubernetes, attach watchers and reconfigure its scraping config 
+based on existing CRs (CustomResource) in Kubernetes.
+
+Changes as well as creations or deletions of said CRs should trigger reconciliation of the scrape config. 
+Active querying of the API should only occur during startup and in a defined interval of the receiver. 
+
+Namespace limitations should be possible for shared cluster scenarios.     
+This should be implemented using industry standard methods. 
+
+
+### Out of scope
+Excluded are other relevant CRDs such as: 
+- PrometheusRules
+- ThanosRuler
+- Probes
+- Alertmanagers
+- AlertmanagerConfigs
+
+This includes concepts like alerting and black box probing of defined targets. 
+
+
+## PrometheusOperator CR watcher
+The CR watcher is the component responsible for querying CRs from Kubernetes API and triggering a reconciliation.
+
+### Major components of CR watcher
+- **[ListenerFactory](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/informers/monitoring.go):**
+  the component which creates listeners on the CR
+- **[APIClient](https://github.com/prometheus-operator/prometheus-operator/tree/main/pkg/client):**
+  PrometheusOperator API client
+
+## Config generator
+The config generator is triggered by the watcher and generates a Prometheus config as a byte array.   
+Instead of writing it onto a disk the configuration is unmarshalled using the Prometheus config loader, 
+which is already in use by the `prometheusreceiver` resulting in a full Prometheus config. 
+
+In case of an [Agent](#Collector vs Agent deployment) deployment, which is signaled with the `limit_to_node` option, 
```

Comment:
I had 3 limitation options in mind: 
- namespace(s) which are to be watched for monitor objects
- a label selector to limit monitor objects in these namespaces( as it is setup in the Prometheus CRD of PrometheusOperator )
- the node limiter option, which is used for an agent style deployment. 

The first two are currently provided by PrometheusOperator `ConfigGenerator` package and the later one is implemented using the additional relabel configs.

Reply:
I'm just suggesting that we re-use the underlying prometheus [namespaces and selector](https://github.com/prometheus/prometheus/blob/main/discovery/kubernetes/kubernetes.go#L123) structure.  It allows specifying the role (podmonitor or servicemonitor in this case), a label selector, and a field selector.   The field selector would allow limiting to podmonitors or servicemonitors on the same node, but is more general than your proposed node limiter option.  Because of the "role" field, it would also allow supporting only podmonitors, or only servicemonitors, and allows different label or field selectors for each.  Re-using the prometheus server's structure for this config would make it familiar to those already familiar with kubernetes_sd_configs.